pypesto/objective/amici/amici.py

```diff
@@ -599,8 +603,13 @@ def set_custom_timepoints(
                 for _ in range(len(amici_objective.edatas))
             ]
 
-        amici_objective.custom_timepoints = custom_timepoints
-        amici_objective.apply_custom_timepoints()
+        if type(amici_objective.calculator) is HierarchicalAmiciCalculator:
+            amici_objective.calculator.set_simulation_edatas(
+                amici_objective.edatas, custom_timepoints
+            )
+        else:
+            amici_objective.custom_timepoints = custom_timepoints
+            amici_objective.apply_custom_timepoints()
```

Comment:
I strongly prefer keeping HierarchicalAmiciCalculator out of here. Could you briefly explain the problem here? The ExpDatas are modified here, and then they are not used by the HierarchicalAmiciCalculator when created copy of the objective is used? This should probably be fixed somewhere else.

Reply:
In general, I put the modifications into the calculator since this was the best way to change the least amount of existing code and it is only needed for the hierarchical setup (so far). Since for a non-hierarchical setup it is okay to use the `apply_custom_timepoints` function from the AmiciObjective, which will change the edatas. 
But for the hierarchical setup, this would change the calculation of inner parameters and is therefore not allowed so far. 

If one would only want the `apply_custom_timepoints` function to work, it should be enough to modify the `check_edatas` function to allow for NaN values that correspond to just having more dense timepoints. 
I did not do that, since I would not know how to precisely check this and the simulation edatas might be valuable for future modification of simulation/visualization routines.

Reply:
In short, if one wants a copy of an objective with more timepoints that is also hierarchical, the edatas is not allowed to change. But then the simulation edatas (the one passed to AMICI after the optimal scaling+offset are calculated) is changed.